src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/KeyworkerService.java

```diff
@@ -153,4 +173,50 @@ public void allocate(OffenderKeyworker allocation) {
     public List<OffenderKeyworker> getAllocationsForKeyworker(Long staffId) {
         return repository.findByStaffId(staffId);
     }
+
+    public Page<KeyworkerDto> getActiveKeyworkers(String agencyId, Optional<String> nameFilter, PagingAndSortingDto pagingAndSorting) {
```

Comment:
This can return Key workers that are not active (if local Key worker record exists and status is not active). I have a similar issue with getting unallocated offenders - it seriously complicates pagination if you start removing non-compliant records before responding so I think the 'decoration' approach is correct but the method would be better named something like 'getKeyworkersWithStatus' (or just 'getKeyworkers'). I would call this from auto-allocate process and just filter out the non-active Key workers.

Reply:
could have sworn the staff search was returning active only! Rename coming up